### ISSUE TITLE
Index error after update RocketChat Version 7.0.0 to 7.0.1 

### ISSUE THREAD

**Issue by thokich** [RocketChat/Rocket.Chat#34132]
### Description:

In the Mongo DB exist in Version 7.0.0 an index which can not changed in the update procedure to version 7.0.1

Some indexes for collection 'rocketchat_user_data_files' could not be created:
Error:
An existing index has the same name as the requested index. When index names are not specified, they are auto generated and can cause conflicts. Please refer to our documentation. Requested index: { v: 2, key: { userId: 1 }, name: "userId_1", sparse: true }, existing index: { v: 2, key: { userId: 1 }, name: "userId_1" }

### Steps to reproduce:

1. Install RocketChat Version 7.0.0 with Mongo in an Docker Container

2. Make an Update to the RocketChat 7.0.1 Container (docker compose update -d)

3. I have in the startup log from the RocketChat Container the Error

Some indexes for collection 'rocketchat_user_data_files' could not be created:
An existing index has the same name as the requested index. When index names are not specified, they are auto generated and can cause conflicts. Please refer to our documentation. Requested index: { v: 2, key: { userId: 1 }, name: "userId_1", sparse: true }, existing index: { v: 2, key: { userId: 1 }, name: "userId_1" }

### Expected behavior:

Normal Startup from the RocketChat Container without this message.
It ist a problem in the Indexmodel in Mogo Database from RocketChat Schema.

### Server Setup Information:

-+---------------------------------------------------------+
|                      SERVER RUNNING                     |
+---------------------------------------------------------+
|                                                         |
|  Rocket.Chat Version: 7.1.0                             |
|       NodeJS Version: 20.18.0 - x64                     |
|      MongoDB Version: 7.0.7                             |
|       MongoDB Engine: wiredTiger                        |
|             Platform: linux                             |
|         Process Port: 3000                              |
|             Site URL: https://rocketchat.xxxx.de  |
|     ReplicaSet OpLog: Enabled                           |
|          Commit Hash: 73af9c5375                        |
|        Commit Branch: HEAD                              |
|                                                         |
+---------------------------------------------------------+

### Relevant logs:

Full Startup Process

strict mode: use allowUnionTypes to allow union type keyword at "#/properties/count" (strictTypes)
strict mode: use allowUnionTypes to allow union type keyword at "#/properties/offset" (strictTypes)
(node:1) NOTE: The AWS SDK for JavaScript (v2) is in maintenance mode.
 SDK releases are limited to address critical bug fixes and security issues only.
Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the blog post at https://a.co/cUPnyil
(Use `node --trace-warnings ...` to show where the warning was created)
Some indexes for collection 'rocketchat_user_data_files' could not be created:
	An existing index has the same name as the requested index. When index names are not specified, they are auto generated and can cause conflicts. Please refer to our documentation. Requested index: { v: 2, key: { userId: 1 }, name: "userId_1", sparse: true }, existing index: { v: 2, key: { userId: 1 }, name: "userId_1" }
LocalStore: store created at 
LocalStore: store created at 
LocalStore: store created at 
2024-12-06T13:46:47.604Z 40 pid=1 hostname=2e57b0365cd5 name=VoIPAsteriskService msg=Voip is not enabled. Cant start the service
2024-12-06T13:46:47.858Z 51 pid=1 hostname=2e57b0365cd5 name=Migrations msg=Not migrating, already at version 318
[2024-12-06T13:46:47.892Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/BROKER: Moleculer v0.14.35 is starting...
[2024-12-06T13:46:47.907Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/BROKER: Namespace: <not defined>
[2024-12-06T13:46:47.907Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/BROKER: Node ID: e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d
[2024-12-06T13:46:47.913Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/REGISTRY: Strategy: RoundRobinStrategy
[2024-12-06T13:46:47.915Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/REGISTRY: Discoverer: LocalDiscoverer
[2024-12-06T13:46:47.918Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/BROKER: Serializer: EJSONSerializer
[2024-12-06T13:46:47.931Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/BROKER: Validator: FastestValidator
[2024-12-06T13:46:47.933Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/BROKER: Registered 13 middleware(s).
[2024-12-06T13:46:47.934Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/BROKER: Transporter: TcpTransporter
2024-12-06T13:46:47.970Z 51 pid=1 hostname=2e57b0365cd5 name=DatabaseWatcher msg=Using change streams
[2024-12-06T13:46:47.978Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/TRANSIT: Connecting to the transporter...
[2024-12-06T13:46:47.994Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/TRANSPORTER: TCP server is listening on port 33759
[2024-12-06T13:46:47.997Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/TRANSPORTER: UDP Discovery is disabled.
[2024-12-06T13:46:47.999Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/TRANSPORTER: TCP Transporter started.
2024-12-06T13:46:48.750Z 51 pid=1 hostname=2e57b0365cd5 name=License msg=License installed version=3.0 hash=dmcMGo9A
[2024-12-06T13:46:48.975Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/REGISTRY: '$node' service is registered.
[2024-12-06T13:46:48.977Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/REGISTRY: 'matrix' service is registered.
[2024-12-06T13:46:48.977Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/$NODE: Service '$node' started.
[2024-12-06T13:46:48.977Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/MATRIX: Service 'matrix' started.
[2024-12-06T13:46:48.978Z] INFO  e74b1b66-9d09-4ce7-9ebd-5faf8e1bc57d/BROKER: ✔ ServiceBroker with 2 service(s) started successfully in 1s.

ufs: temp directory created at "/tmp/ufs"

+---------------------------------------------------------+
|                      SERVER RUNNING                     |
+---------------------------------------------------------+
|                                                         |
|  Rocket.Chat Version: 7.1.0                             |
|       NodeJS Version: 20.18.0 - x64                     |
|      MongoDB Version: 7.0.7                             |
|       MongoDB Engine: wiredTiger                        |
|             Platform: linux                             |
|         Process Port: 3000                              |
|             Site URL: https://rocketchat.xxxx.de  |
|     ReplicaSet OpLog: Enabled                           |
|          Commit Hash: 73af9c5375                        |
|        Commit Branch: HEAD                              |
|                                                         |
+---------------------------------------------------------+

2024-12-06T13:46:50.332Z 51 pid=1 hostname=xxx name=License msg=License installed version=3.0 hash=xxx

**Comment by thokich** [comment]
Hey, i have look in Mongo with mongo express an hab´ve made a screenshot from dthe index from the collection rocketchat_user_data_files.
![Bildschirmfoto 2024-12-06 um 15 29 12](https://github.com/user-attachments/assets/b4e14ccb-0fbf-4823-8429-d2e40a544681)

**Comment by thokich** [comment]
Hey, can i do an Index workaroud on mongo-db ? :-)
db.collectionName.dropIndex("indexName")
db.collectionName.createIndex({ feld1: 1, feld2: -1 })

**Comment by mremptypants** [comment]
Yeah getting the same error now..if users signed out, they can not sign in.

**Comment by thokich** [comment]
Signout/SignIn with the Browser and iOS App Login works in my enviroment.

**Comment by reetp** [comment]
Just drop the indexes.

The system will create as required.

**Comment by thokich** [comment]
hey, i have test an initial setup. 
1. I setup an empty mongodb in an container (docker compose...)
2. i setup rocketchat in an container (docker compose..)
3. Index failure in startup when i make an initial setup !!

trict mode: use allowUnionTypes to allow union type keyword at "#/properties/count" (strictTypes)
strict mode: use allowUnionTypes to allow union type keyword at "#/properties/offset" (strictTypes)
(node:1) NOTE: The AWS SDK for JavaScript (v2) is in maintenance mode.
 SDK releases are limited to address critical bug fixes and security issues only.
Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the blog post at https://a.co/cUPnyil
(Use `node --trace-warnings ...` to show where the warning was created)
Some indexes for collection 'rocketchat_user_data_files' could not be created:

	An existing index has the same name as the requested index. When index names are not specified, they are auto generated and can cause conflicts. Please refer to our documentation. Requested index: { v: 2, key: { userId: 1 }, name: "userId_1" }, existing index: { v: 2, key: { userId: 1 }, name: "userId_1", sparse: true }

LocalStore: store created at 
LocalStore: store created at 
LocalStore: store created at 
2024-12-09T09:39:01.430Z 40 pid=1 hostname=4b78f216a426 name=VoIPAsteriskService msg=Voip is not enabled. Cant start the service
[2024-12-09T09:39:09.585Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/BROKER: Moleculer v0.14.35 is starting...
[2024-12-09T09:39:09.586Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/BROKER: Namespace: <not defined>
[2024-12-09T09:39:09.586Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/BROKER: Node ID: 3a2118d0-1c8c-4620-aeb9-05e62d30a9c4
[2024-12-09T09:39:09.589Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/REGISTRY: Strategy: RoundRobinStrategy
[2024-12-09T09:39:09.590Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/REGISTRY: Discoverer: LocalDiscoverer
[2024-12-09T09:39:09.592Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/BROKER: Serializer: EJSONSerializer
[2024-12-09T09:39:09.618Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/BROKER: Validator: FastestValidator
[2024-12-09T09:39:09.624Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/BROKER: Registered 13 middleware(s).
[2024-12-09T09:39:09.631Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/BROKER: Transporter: TcpTransporter
2024-12-09T09:39:09.696Z 51 pid=1 hostname=4b78f216a426 name=DatabaseWatcher msg=Using change streams
[2024-12-09T09:39:09.704Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/TRANSIT: Connecting to the transporter...
[2024-12-09T09:39:09.743Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/TRANSPORTER: TCP server is listening on port 37539
[2024-12-09T09:39:09.744Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/TRANSPORTER: UDP Discovery is disabled.
[2024-12-09T09:39:09.745Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/TRANSPORTER: TCP Transporter started.
ufs: temp directory created at "/tmp/ufs"
[2024-12-09T09:39:10.328Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/REGISTRY: '$node' service is registered.
[2024-12-09T09:39:10.330Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/REGISTRY: 'matrix' service is registered.
[2024-12-09T09:39:10.331Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/$NODE: Service '$node' started.
[2024-12-09T09:39:10.332Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/MATRIX: Service 'matrix' started.
[2024-12-09T09:39:10.332Z] INFO  3a2118d0-1c8c-4620-aeb9-05e62d30a9c4/BROKER: ✔ ServiceBroker with 2 service(s) started successfully in 658ms.
+---------------------------------------------------+
|                   SERVER RUNNING                  |
+---------------------------------------------------+
|                                                   |
|  Rocket.Chat Version: 7.1.0                       |
|       NodeJS Version: 20.18.0 - x64               |
|      MongoDB Version: 7.0.14                      |
|       MongoDB Engine: wiredTiger                  |
|             Platform: linux                       |
|         Process Port: 3000                        |
|             Site URL: https://rocketchat.demo.de  
|     ReplicaSet OpLog: Enabled                     |
|          Commit Hash: 73af9c5375                  |
|        Commit Branch: HEAD                        |
|                                                   |
+---------------------------------------------------+

**Comment by reetp** [comment]
Not exactly sure why I got a thumbs down?

I'm stating a fact. Normally if you have a duplicate index you can just drop them and the system will re-create as required.

You also gave yourself a thumbs down. Perhaps you ought to go back and undo those?

> Can you send me the right create index statement ?

You don't need one as per the above. They system attempts to create them itself.

> I see no collection 'rocketchat_user_data_files'

Likely the issue. I think this collection holds information when a user requests a copy of their personal information. So it is probably nor created until first use. And so it will complain about failing to create an index on a non-existent collection.

Effectively it is log noise but I will refer it to the team, though I don't expect it will be high on the priority list.

NB: if you see this your server is essentially up and running happily.

```
+---------------------------------------------------+
| SERVER RUNNING |
+---------------------------------------------------+
| |
| Rocket.Chat Version: 7.1.0 |

```

**Comment by thokich** [comment]
Yes, the Server is running and working. Thats great. I am an Rocket Fan.
Thanks for your fast reaction.
Let us see what the Development do. I am an a Developer, too and i know this problems :-)

**Comment by thokich** [comment]
Thats funny.
I have create the Index manually and restart rocket container

db.rocketchat_user_data_files.createIndex(
    { userId: 1 }, // Index-Schlüssel
    { name: "userId_1", sparse: true, v: 2 } // Optionen
  );


Than the output is:

Some indexes for collection 'rocketchat_user_data_files' could not be created:
	An existing index has the same name as the requested index. When index names are not specified, they are auto generated and can cause conflicts. Please refer to our documentation. Requested index: { v: 2, key: { userId: 1 }, name: "userId_1" }, existing index: { v: 2, key: { userId: 1 }, name: "userId_1", sparse: true }
LocalStore: store created at 

what ?
sparse: true or not :-)

Info from top:
Please refer to our documentation. Requested index: { v: 2, key: { userId: 1 }, name: "userId_1", sparse: true }, existing index: { v: 2, key: { userId: 1 }, name: "userId_1" }

**Comment by thokich** [comment]
![Uploading Bildschirmfoto 2024-12-09 um 12.50.32.png…]()

**Comment by thokich** [comment]
![Bildschirmfoto 2024-12-09 um 12 50 32](https://github.com/user-attachments/assets/6c9ce8ce-39ab-4381-b4bd-cc98f3031801)

**Comment by thokich** [comment]
Orginalindexes after Migration for this collection

{"indexes":[
{"v":{"$numberInt":"2"},"key":{"_id":{"$numberInt":"1"}},"name":"_id_"},
{"v":{"$numberInt":"2"},"key":{"rid":{"$numberInt":"1"}},"name":"rid_1","sparse":true},
{"v":{"$numberInt":"2"},"key":{"userId":{"$numberInt":"1"}},"name":"userId_1"},
{"v":{"$numberInt":"2"},"key":{"name":{"$numberInt":"1"}},"name":"name_1","sparse":true},
{"v":{"$numberInt":"2"},"key":{"expiresAt":{"$numberInt":"1"}},"name":"expiresAt_1","sparse":true}],"
uuid":"e3e15cfc88e64864b001de96d71af24c","collectionName":"rocketchat_user_data_files","type":"collection"}

**Comment by reetp** [comment]
> I have create the Index manually and restart rocket container

You do not need to do that - the system does it automatically.

As I have described this is likely an error due to the collection not being created initially. It is probably only created on first use. But the system is trying to create an index immediately.

If you try and export user data you will probably find the collection is created and you can then drop the index on that collection, restart and let the server create new indexes.

There is nothing much else to do here -  I have reported it o the team.

**Comment by thokich** [comment]
I don't think so.
We have two Bug Szenarios in my issues.

1. Upgrade from Rocket Version 7.0.0 to 7.1.0 
2. Initial Setup from zero to Version 7.1.0

Possible two Bugs one cause. first the Index Creation and second the Collation creation in initial setup. I belive this is a Failure from the Index Creation BUG and then the Collation was not create.

Only the Code says the True :-)

**Comment by reetp** [comment]
> We have two Bug Scenarios in my issues

I have already told you the likely underlying cause. The collection not being created on installation. And the index fails to be created thereafter.

 I have also referred it to the development team so there is no need to comment endlessly about it. See what they say.

Of course if you have a solution they will be happy to look at your PR.

Thanks.

**Comment by casalsgh** [comment]
Will share with internal teams
